spec/models/contribution_spec.rb

```diff
@@ -53,12 +45,21 @@
     end
   end
 
-  describe '#authorship_valid?' do
-    it 'calls #author_valid?' do
-      expect(described_class).to receive(:author_valid?)
-      described_class.authorship_valid?(authorship)
+  context 'ActiveRecord validation' do
+    let(:params) { { publication_id: pub_with_contrib.id, author_id: subject.author.id, visibility: 'private', status: 'approved' } }
+    it 'constrains field values' do
+      expect { described_class.create!(params) }.not_to raise_error
+      expect { described_class.create! }.to raise_error ActiveRecord::RecordInvalid
+      expect { described_class.create!(params.merge(visibility: 'bad'))   }.to raise_error ActiveRecord::RecordInvalid
+      expect { described_class.create!(params.merge(status: 'illegal'))   }.to raise_error ActiveRecord::RecordInvalid
+      expect { described_class.create!(params.merge(author: nil))         }.to raise_error ActiveRecord::RecordInvalid
+      expect { described_class.create!(params.merge(publication_id: nil)) }.to raise_error ActiveRecord::RecordInvalid
     end
```

Comment:
My IDE always complains about more than one expectation per spec, but this is all related stuff.  I suppose the best practice is to have one expect per spec, so they can't interfere or short-circuit one another.  For example, if line 52 were to fail, nothing else would execute.  I think that when you know all the expectations build on one another, like stepping stones, it makes sense to combine them and it doesn't matter if the spec short-circuits.  In this spec, I'm wondering if they are dependent or independent specs.  I could see an argument for separating the specs for each field.  If your open to separating them, that might be better.

Reply:
I much prefer these as is.  It would be different if they were performing operations on a common object, but they are independent block executions on the constructor that raise exceptions, so there is literally no way for them to side-effect each other.  They are testing permutations of the same conditions, so giving them separate descriptions is unnecessary, imo.

Reply:
Do you care if the spec execution stops at some point and does not test the others?  If it does stop, hopefully the line number is sufficient to be able to track which one fails.  This thing is no big deal, IMO, just noting an option to pursue best practice that recommends one-expect per spec.